src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -296,6 +298,9 @@ class DashboardWidgetSerializer(CamelSnakeSerializer[Dashboard]):
     limit = serializers.IntegerField(min_value=1, max_value=10, required=False, allow_null=True)
     layout = LayoutField(required=False, allow_null=True)
     query_warnings: QueryWarning = {"queries": [], "columns": {}}
+    dataset_source = serializers.ChoiceField(
```

Comment:
since this is a private field, you should add the `extend_schema_serializer` decorator (see [docs](https://develop.sentry.dev/application/api/public/#5-method-decorator))

that being said, the original issue with the test failure has to do with `allow_null=True`. You can change this to `allow_blank=True` (the preferred for textual choices) and it shouldn't reproduce the error.